infrastructure/src/main/java/org/corfudb/infrastructure/redundancy/RedundancyCalculator.java

```diff
@@ -0,0 +1,218 @@
+package org.corfudb.infrastructure.redundancy;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Sets;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NonNull;
+import org.corfudb.infrastructure.log.statetransfer.StateTransferManager.TransferSegment;
+import org.corfudb.runtime.view.Layout;
+import org.corfudb.runtime.view.Layout.LayoutSegment;
+import org.corfudb.runtime.view.Layout.LayoutStripe;
+import org.corfudb.runtime.view.LayoutBuilder;
+
+import java.util.List;
+import java.util.function.IntPredicate;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import static org.corfudb.infrastructure.log.statetransfer.StateTransferManager.TransferSegmentStatus;
+import static org.corfudb.infrastructure.log.statetransfer.StateTransferManager.TransferSegmentStatus.SegmentState.NOT_TRANSFERRED;
+import static org.corfudb.infrastructure.log.statetransfer.StateTransferManager.TransferSegmentStatus.SegmentState.RESTORED;
+import static org.corfudb.runtime.view.Address.NON_ADDRESS;
+
+/**
+ * A class used to compute the transfer segments, as well as the layout
+ * after the redundancy restoration.
+ */
+@AllArgsConstructor
+public class RedundancyCalculator {
+
+    @NonNull
+    @Getter
+    private final String server;
+
+    boolean segmentContainsServer(LayoutSegment segment, String server) {
+        return segment.getFirstStripe().getLogServers().contains(server);
+    }
+
+
+    /**
+     * Given a current layout and a restored transfer segment, create a new layout, that
+     * contains a restored node.
+     *
+     * @param transferSegment A restored transfer segment.
+     * @param layout          A current layout.
+     * @return A new, updated layout.
+     */
+    Layout restoreRedundancyForSegment(TransferSegment transferSegment, Layout layout) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use explicit scoping instead of the default package private level](https://app.codacy.com/app/annymartinez.m/CorfuDB/pullRequest?prid=4651453)